src/legacy/ui/public/kfetch/kfetch.ts

```diff
@@ -61,10 +62,11 @@ export async function kfetch(
       });
 
       return window.fetch(fullUrl, restOptions).then(async res => {
-        const body = await getBodyAsJson(res);
+        const body = parseJson ? await getBodyAsJson(res) : null;
         if (res.ok) {
-          return body;
+          return parseJson ? body : res;
         }
+
```

Comment:
I'd like to get these changes in soon, but eventually, before we can merge the feature to master, reviewers will want to see a unit test for this change